### PR TITLE
183 changes desc

### PR DESCRIPTION
**Pull request by ricellis**
## What

Added `descending` option to changes feed.

## How
Removed lightcouch `Changes` and `ChangesResult`.
Moved method implementations into cloudant package for `Changes` and `ChangesResult` classes.
Refactored `Changes` constructor and called it from `Database.changes()`
Fixed doc on `continuousChanges` method.
Added descending method to `Changes` class.
Updated CHANGES.md.

## Testing

Added test for changes `descending=true`.

## Reviewers
reviewer @rhyshort
reviewer @emlaver

## Issues
#183

### CODE REVIEW

**Comment by rhyshort** [comment]
+1

File: cloudant-client/src/main/java/com/cloudant/client/api/Changes.java [hunk 1]
```diff
@@ -115,54 +148,104 @@ public void stop() {
...
     public ChangesResult getChanges() {
-        com.cloudant.client.org.lightcouch.ChangesResult couchDbChangesResult = changes
-                .getChanges();
-        ChangesResult changeResult = new ChangesResult(couchDbChangesResult);
-        return changeResult;
+        final URI uri = this.databaseHelper.changesUri("feed", "normal");
+        return client.get(uri, ChangesResult.class);
     }
 
     // Query Params
```
Comment (emlaver) [inline cloudant-client/src/main/java/com/cloudant/client/api/Changes.java]: Wouldn't having java docs (even if it's one sentence) be helpful for the query parameters below?

Comment (ricellis) [inline cloudant-client/src/main/java/com/cloudant/client/api/Changes.java]: That was just a direct code move of the duplicate lightcouch methods into a single class, neither of which was documented. I agree they should be documented though so I have added docs in 088dbcc.

**Comment by emlaver** [comment]
+1

**Pull request merged**